cpp/dolfinx/fem/interpolate.h

```diff
@@ -532,19 +569,33 @@ void interpolate(Function<T>& u, const xt::xarray<T>& f,
     std::span<const double> x_g = mesh->geometry().x();
 
     // Create data structures for Jacobian info
-    xt::xtensor<double, 3> J = xt::empty<double>({int(X.shape(0)), gdim, tdim});
-    xt::xtensor<double, 3> K = xt::empty<double>({int(X.shape(0)), tdim, gdim});
-    xt::xtensor<double, 1> detJ = xt::empty<double>({X.shape(0)});
-
-    xt::xtensor<double, 2> coordinate_dofs
-        = xt::empty<double>({num_dofs_g, gdim});
-
-    xt::xtensor<T, 3> reference_data({X.shape(0), 1, value_size});
-    xt::xtensor<T, 3> _vals({X.shape(0), 1, value_size});
-
-    // Tabulate 1st order derivatives of shape functions at interpolation coords
-    xt::xtensor<double, 3> dphi = xt::view(
-        cmap.tabulate(1, X), xt::range(1, tdim + 1), xt::all(), xt::all(), 0);
+    std::vector<double> J_b(Xshape[0] * gdim * tdim);
+    mdspan3_t J(J_b.data(), Xshape[0], gdim, tdim);
+    std::vector<double> K_b(Xshape[0] * tdim * gdim);
+    mdspan3_t K(K_b.data(), Xshape[0], tdim, gdim);
+    std::vector<double> detJ(Xshape[0]);
+    std::vector<double> det_scratch(2 * gdim * tdim);
+
+    std::vector<double> coord_dofs_b(num_dofs_g * 3);
```

Comment:
Is there any point of assigning `3*num_dofs_g` instead of `gdim*num_dofs_g`, when we will never use the entry 
i>=gdim and results in strided memory access in line 625.

Reply:
No point. Changed.